### PR TITLE
demos: prevent several potential integer overflows

### PR DESCRIPTION
**Pull request by IRDonch**

### CODE REVIEW

File: demos/common/samples/common.hpp [hunk 1]
```diff
@@ -412,60 +412,40 @@ static UNUSED bool writeOutputBmp(std::string name, unsigned char *data, size_t
 * @param width - width of the rectangle
 * @param rectangles - vector points for the rectangle, should be 4x compared to num classes
 * @param classes - vector of classes
-* @param thickness - thickness of a line (in pixels) to be used for bounding boxes
 */
-static UNUSED void addRectangles(unsigned char *data, size_t height, size_t width, std::vector<int> rectangles, std::vector<int> classes, int thickness = 1) {
+static UNUSED void addRectangles(unsigned char *data, size_t height, size_t width, std::vector<int> rectangles, std::vector<int> classes) {
```
Comment (Wovchena) [inline demos/common/samples/common.hpp]: May be `object_detection_demo_faster_rcnn` could just use OpenCV? And the need in this function would be eliminated. :)

File: demos/object_detection_demo_faster_rcnn/detectionoutput.h [hunk 2]
```diff
@@ -107,7 +107,7 @@ class DetectionOutputPostProcessor {
                     {Precision::I32, indices_size, {indices_size, {0, 1, 2}}});
             _indices->allocate();
 
-            SizeVector detections_size{static_cast<size_t>(_num * _num_classes)};
+            SizeVector detections_size{static_cast<size_t>(_num) * static_cast<size_t>(_num_classes)};
```
Comment (Wovchena) [inline demos/object_detection_demo_faster_rcnn/detectionoutput.h]: It looks `_num` is never used as `int`. Except for `_num = static_cast<int>(conf_size);` but `conf_size` is `uint32_t`. `_num` should be defined as `size_t`.

Comment (IRDonch) [reply demos/object_detection_demo_faster_rcnn/detectionoutput.h]: Good catch. `_num` is actually entirely redundant, so I just removed it.

File: demos/common/samples/common.hpp (hunk 1 above)
Comment (IRDonch) [reply demos/common/samples/common.hpp]: Fair point. I'll see if it's not too difficult to port this demo to OpenCV.

Comment (IRDonch) [reply demos/common/samples/common.hpp]: Ported it in #606. This change no longer applies, so I removed it.

**Pull request merged**